src/AppInstallerRepositoryCore/Microsoft/PinningIndex.cpp

```diff
@@ -74,7 +91,7 @@ namespace AppInstaller::Repository::Microsoft
     PinningIndex::IdType PinningIndex::AddPin(const Pinning::Pin& pin)
     {
         std::lock_guard<std::mutex> lockInterface{ *m_interfaceLock };
-        AICLI_LOG(Repo, Verbose, << "Adding Pin for package [" << pin.GetPackageId() << "] from source [" << pin.GetSourceId() << "] with pin type " << Pinning::ToString(pin.GetType()));
+        AICLI_LOG(Repo, Verbose, << "Adding Pin for " << GetPinKeyLogString(pin.GetKey()) << " with pin type " << Pinning::ToString(pin.GetType()));

```

Comment:
If I'm not mistaken, the only time that the PinKey or the PinType need to be converted to a string is for logging. I think this may be better if you overload `Pinning::ToString` with `Pinning::ToString(PinKey)` or maybe even `Pinning::ToString(Pin)`.

You could even simplify the `with pin type` to be part of the string as `Type=[ ]` just to keep it all the same format